onnxruntime/core/providers/cpu/ml/ml_common.h

```diff
@@ -254,58 +254,62 @@ static inline float sigmoid_probability(float score, float proba, float probb) {
   return 1 - ComputeLogistic(val);  // ref: https://github.com/arnaudsj/libsvm/blob/eaaefac5ebd32d0e07902e1ae740e038eaaf0826/svm.cpp#L1818
 }
 
-static inline void ComputeSoftmax(const gsl::span<float>& values) {
+template <typename T>
+static inline void ComputeSoftmax(const gsl::span<T>& values) {
   // TODO: Replace this with usage of code in Softmax operator
 
   // compute exp with negative number to be numerically stable
   float v_max = -std::numeric_limits<float>::max();
-  for (float value : values) {
-    if (value > v_max)
-      v_max = value;
+  for (auto it = values.cbegin(); it != values.cend(); ++it) {
+    if (static_cast<float>(*it) > v_max)
+      v_max = static_cast<float>(*it);
   }
```

Comment:
Curious as to what differences there are in the assembly generated for the old vs. new code as I would have expected it to be the same in a release build.

Reply:
The difference is this code can handle other type than float, which I needed in this case.